fetch-publication/src/test/java/no/unit/nva/publication/fetch/FetchPublicationHandlerTest.java

```diff
@@ -130,9 +135,9 @@ public void handlerReturnsInternalServerErrorResponseOnUnexpectedException()
         when(publicationService.getPublication(any(UUID.class)))
             .thenThrow(new NullPointerException());
 
-        fetchPublicationHandler.handleRequest(inputStream(), output, context);
+        fetchPublicationHandler.handleRequest(generateHandlerRequest(), output, context);
 
-        GatewayResponse gatewayResponse = objectMapper.readValue(output.toString(), GatewayResponse.class);
+        GatewayResponse<PublicationResponse> gatewayResponse = parseHandlerResponse();
```

Comment:
Use the methods in the first suggestion and fix the tests according to the suggestions. On error we do not expect PublicationResponse. We expect a Problem. Unfortunately, jackson's javatype does not throw error when the generic type is wrong. So we have to use the Problem instance (i.e., call the Problem::getDetails()) method in order to verify that the type is indeed a problem with the desired details